### PR TITLE
Add installation guide

### PR DESCRIPTION
**Pull request by VoVAllen**
Signed-off-by: Jinjing.Zhou <allenzhou@tensorchord.ai>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,63 +8,39 @@
...
-
-## Setting up the development environment
-
-You could use [envd](https://github.com/tensorchord/envd) to set up the development environment with one command. It will create a docker container and install all the dependencies for you.
+### Install Rust and base dependency
 
 ```sh
-pip install envd
-envd up
+apt install -y build-essential libpq-dev libssl-dev pkg-config gcc zlib1g-dev
```
Comment (usamoi) [inline README.md:26]: libreadline-dev flex bison libxml2-dev libxslt-dev libxml2-utils xsltproc ccache clang

File: README.md [hunk 2]
```diff
@@ -158,6 +134,46 @@ We planning to support more index types ([issue here](https://github.com/tensorc
...
+## Setting up the development environment
+
+You could use [envd](https://github.com/tensorchord/envd) to set up the development environment with one command. It will create a docker container and install all the dependencies for you.
+
+```sh
+pip install envd
+envd up
+```
+
+## Build from source
```
Comment (usamoi) [inline README.md:169]: Is this section is replicated?

Comment (VoVAllen) [reply README.md]: Removed

File: README.md (hunk 1 above)
Comment (VoVAllen) [reply README.md]: Added without ccache and clang, hence I added gcc already

Comment (usamoi) [reply README.md]: @VoVAllen Crate `bindgen` needs `clang-sys`. It's necessary. I'm not sure whether `ccache` is needed. The [link](https://wiki.postgresql.org/wiki/Compile_and_Install_from_source_code) says yes.

Comment (VoVAllen) [reply README.md]: added

**Pull request merged**